.eslintrc.js

```diff
@@ -30,17 +39,32 @@ module.exports = {
                 "import/no-commonjs": "off",
             },
         },
+        {
+            files: ["**/*.test.ts", "**/*.test.tsx"],
+            rules: {
+                "no-undef": "off",
+            },
+        },
     ],
     globals: {
         // `no-undef` doesn't support `globalThis`, for details see
         // https://github.com/eslint/eslint/issues/15199.
         globalThis: false, // means it isn't writeable
+        // from node_modules/@types/react/index.d.ts
+        JSX: false,
+        // from node_modules/typescript/lib/lib.dom.d.ts
+        DOMHighResTimeStamp: false,
+        RequestInfo: false,
+        RequestInit: false,
+        // from types/utility.d.ts
+        Empty: false,
+        ObjMap: false,
+        SpreadType: false,
```

Comment:
Some people suggest disable `no-undef` all the time, e.g. https://stackoverflow.com/questions/65054079/eslint-with-typescript-and-firebase-no-undef-error.  That doesn't seem like a good solution.  Eslint should be able to tell the difference between value identifier and type identifiers and ignore just the undefined type identifiers.